cmd/benchdiff/internal/gitrunner.go

```diff
@@ -16,57 +18,27 @@ func runGitCmd(debug *log.Logger, gitCmd, repoPath string, args ...string) ([]by
 	return bytes.TrimSpace(stdout.Bytes()), err
 }
 
-func stashAndReset(debug *log.Logger, gitCmd, repoPath string) (revert func() error, err error) {
-	revert = func() error {
-		return nil
-	}
-	stash, err := runGitCmd(debug, gitCmd, repoPath, "stash", "create")
-	if err != nil {
-		return nil, err
-	}
-	stash = bytes.TrimSpace(stash)
-	if len(stash) > 0 {
-		revert = func() error {
-			_, revertErr := runGitCmd(debug, gitCmd, repoPath, "stash", "apply", string(stash))
-			return revertErr
-		}
-	}
-	_, err = runGitCmd(debug, gitCmd, repoPath, "reset", "--hard")
-	if err != nil {
-		return nil, err
-	}
-	return revert, nil
-}
-
-func runAtGitRef(debug *log.Logger, gitCmd, repoPath, ref string, fn func()) error {
-	origRef, err := runGitCmd(nil, gitCmd, repoPath, "rev-parse", "--abbrev-ref", "HEAD")
+func runAtGitRef(debug *log.Logger, gitCmd, repoPath, ref string, fn func(path string)) error {
+	worktree, err := ioutil.TempDir("", "bindiff")
 	if err != nil {
 		return err
 	}
-	origRef = bytes.TrimSpace(origRef)
-	unstash, err := stashAndReset(debug, gitCmd, repoPath)
-	if err != nil {
-		return err
-	}
-	defer func() {
-		unstashErr := unstash()
-		if unstashErr != nil {
-			panic(unstashErr)
-		}
-	}()
-	_, err = runGitCmd(debug, gitCmd, repoPath, "checkout", "--quiet", ref)
+	defer os.RemoveAll(worktree)
```

Comment:
So they know what can be cleaned up manually if something goes wrong.

```suggestion
	defer func() {
		rErr := os.RemoveAll(worktree)
		if rErr != nil {
			fmt.Printf("Could not delete temp directory: %s\n", worktree)
		}
	}()
```